lib/teleterm/gateway/gateway.go

```diff
@@ -129,6 +137,19 @@ func (g *Gateway) LocalPortInt() int {
 	return port
 }
 
+func (g *Gateway) SetTargetSubresourceName(targetSubresourceName string) error {
+	g.TargetSubresourceName = targetSubresourceName
```

Comment:
One unresolved problem is that `TargetSubresourceName` is still a public field, so it can be changed without updating `CLICommand`.

I decided to leave it as it is for now, as `TargetSubresourceName` is defined on `Config` and I'd need to refactor even more code to make it work, I was worried that it'd make the PR even more chaotic than it already is.